core/variant/variant_construct.h

```diff
@@ -153,11 +153,14 @@ class VariantConstructor {
 class VariantConstructorObject {
 public:
 	static void construct(Variant &r_ret, const Variant **p_args, Callable::CallError &r_error) {
-		VariantInternal::clear(&r_ret);
 		if (p_args[0]->get_type() == Variant::NIL) {
+			VariantInternal::clear(&r_ret);
+			VariantTypeChanger<Object *>::change(&r_ret);
 			VariantInternal::object_assign_null(&r_ret);
```

Comment:
Is this redundant in this case? `VariantTypeChanger<Object *>::change` calls `VariantInitializer<T>::init(v)` and for `Object` that calls `VariantInternal::init_object(v)` which finally calls `object_assign_null` and also sets the type to Object.

Reply:
Nope, see the description, it's about the *old* data, not the new

Reply:
I tried it without several of these and arrived on that this was necessary, as explained above

Reply:
The old data is null. No point in calling `object_assign_null` twice.

Reply:
Please read this https://github.com/godotengine/godot/pull/90134#issuecomment-2032399824

The old data isn't null *without* `clear`, the problem is that we both need to *clear* and then initialize it properly, again please read the original post and my other comments that explain this

Reply:
Thank you for trying to help, it's appreciated, but as I said I've tried different solutions and they didn't work, and I think you need to re-read what the actual problem here is, it's not just the problem in #91289